tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -317,15 +327,51 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub fn get_id_graph(who: &T::AccountId) -> Vec<(Identity, IdentityContext<T>)> {
+		fn insert_identity_with_limit(
+			owner: &T::AccountId,
+			identity: &Identity,
+			context: IdentityContext<T>,
+		) -> Result<(), DispatchError> {
+			IDGraphLens::<T>::try_mutate(owner, |len| {
+				let new_len = len.checked_add(1).ok_or(Error::<T>::IDGraphLenLimitReached)?;
+				if new_len > T::MaxIDGraphLength::get() {
+					return Err(Error::<T>::IDGraphLenLimitReached.into())
+				}
+				*len = new_len;
+				Result::<(), DispatchError>::Ok(())
+			})?;
+			IDGraphs::<T>::insert(owner, identity, context);
+			Ok(())
+		}
```

Comment:
`IDGraphs::<T>::insert` could be put inside the `try_mutate` block to save a `?` and return value :)

Reply:
Maybe, but for clarity I prefer to put in `try_mutate` only the logic related to mutating that `Map`, not any other.

Reply:
Yes, I misread it sorry